src/pyadjoint/adjoint_source.py

```diff
@@ -140,6 +146,41 @@ def _write_specfem(self, buf, time_offset):
 
         np.savetxt(buf, to_write)
 
+    def write_to_asdf(self, ds, time_offset):
+          """
+          Writes the adjoint source to an ASDF file.
+          Note: For now it is assumed SPECFEM will be using the adjoint source
+
+          :param ds: The ASDF data structure read in using pyasdf
+          :type ds: str
+          """
+          # Convert the adjoint source to SPECFEM format
+          l = len(self.adjoint_source)
+          specfem_adj_source = np.empty((l,2))
+          specfem_adj_source[:, 0] = np.linspace(0, (l - 1) * self.dt, l)
+          specfem_adj_source[:, 1] += time_offset
+          specfem_adj_source[:, 1] += self.adjoint_source[::-1]
```

Comment:
This is fixed Thanks for point it it out.

On Oct 21, 2015, at 3:30 AM, Lion Krischer <notifications@github.com<mailto:notifications@github.com>> wrote:


In src/pyadjoint/adjoint_source.py<https://github.com/krischer/pyadjoint/pull/1#discussion_r42591200>:

> @@ -140,6 +146,41 @@ def _write_specfem(self, buf, time_offset):
>
>          np.savetxt(buf, to_write)
>
> +    def write_to_asdf(self, ds, time_offset):
> +          """
> +          Writes the adjoint source to an ASDF file.
> +          Note: For now it is assumed SPECFEM will be using the adjoint source
> +
> +          :param ds: The ASDF data structure read in using pyasdf
> +          :type ds: str
> +          """
> +          # Convert the adjoint source to SPECFEM format
> +          l = len(self.adjoint_source)
> +          specfem_adj_source = np.empty((l,2))
> +          specfem_adj_source[:, 0] = np.linspace(0, (l - 1) * self.dt, l)
> +          specfem_adj_source[:, 1] += time_offset
> +          specfem_adj_source[:, 1] += self.adjoint_source[::-1]


For the two preceding lines: You cannot savely use +=. You used np.empty() to create the array which just grabs a piece of memory with whatever may be in it (might be zeros, might be not). Best replace the += with a normal =.

—
Reply to this email directly or view it on GitHub<https://github.com/krischer/pyadjoint/pull/1/files#r42591200>.